base/linalg/schur.jl

```diff
@@ -107,6 +107,7 @@ function schur(A::StridedMatrix)
     SchurF = schurfact(A)
     SchurF[:T], SchurF[:Z], SchurF[:values]
 end
+schur(A::AbstractMatrix) = schur(full(A))
```

Comment:
The problem with this definition is that it will produce an infinite loop for any `AbstractMatrix` type where `full(A) === A` and there is no more-specific `schur` method.

I would just add `schur(A::Triangular{<:Complex}) = schur(full(A))` (and anything else as needed).